numba/tests/test_utils.py

```diff
@@ -0,0 +1,87 @@
+import os
+import sys
+import threading
+from numba.misc import utils
+from numba.tests.support import TestCase
+from numba import njit
+
+# Solution for capturing stdout
+# https://stackoverflow.com/questions/24277488/in-python-how-to-capture-the-stdout-from-a-c-shared-library-to-a-variable
+# stdout_fileno = sys.stdout.fileno()
+# stdout_save = os.dup(stdout_fileno)
+# stdout_pipe = os.pipe()
+# os.dup2(stdout_pipe[1], stdout_fileno)
+# os.close(stdout_pipe[1])
+
+# captured_stdout = b''
+# def drain_pipe():
+# 	global captured_stdout
+# 	while True:
+# 		data = os.read(stdout_pipe[0], 1024)
+# 		if not data:
+# 			break
+# 		captured_stdout += data
+
+
+# def printf():
+# 	d = i = 392
+# 	utils.printf("d=%d, i=%i\n", d, i)
```

Comment:
there are no tests for %d and %c. is it because they are covered by %i and %s?

Reply:
There are tests for %d and %c. Former is in the text where you highlighted and %c is below.

Reply:
sorry, I didn't see it!